lookout/style/format/analyzer.py

```diff
@@ -43,6 +45,54 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
         :param data: Contains "changes" - the list of changes in the pointed state.
         :return: List of comments.
         """
+        def get_error_description(vnode: VirtualNode, correct_y: int) -> str:
+            """
+            Writes the comment with regard to the correct node class.
+            :param correct_y: The index of the correct node class.
+            :return: String comment.
+            """
+            column = vnode.start.col
+            if correct_y == CLASS_INDEX[CLS_NOOP]:
+                return "`%s` at column %d should be removed." % (CLASSES[vnode.y], column)
+            elif vnode.y == CLASS_INDEX[CLS_NOOP]:
+                return "`%s` should be inserted at column %d." % (CLASSES[correct_y], column)
+            else:
+                return "Replace `%s` with `%s` at column %d." % (
+                    CLASSES[vnode.y], CLASSES[correct_y], column)
+
+        def group_line_nodes(y, y_pred, vnodes_y, rule_winners):
+            line_nodes = []
+            generate_comment = False
+            for yi, y_predi, vnode_y, winner in zip(y, y_pred, vnodes_y, rule_winners):
+                if not line_nodes or vnode_y.start.line == line_nodes[0][2].start.line:
+                    # collect all nodes on the same line
+                    line_nodes.append((yi, y_predi, vnode_y, winner))
+                    if yi != y_predi:
+                        generate_comment = True
+                    continue
+                else:
+                    if generate_comment:
+                        yield line_nodes
+                    generate_comment = yi != y_predi
+                    line_nodes = [(yi, y_predi, vnode_y, winner)]
+
+        def get_code_chunk(lang, code_lines, line_number):
+            lines = list(range(max(0, line_number - 2), line_number + 1))
```

Comment:
should be line_number + 2 as stopping condition so that the line is in the middle.

Reply:
no, because `line_number` is 1-based here, so it is:
 ```
line_number -= 1
list(range(max(0, line_number - 1), line_number + 2))`

Reply:
but there are only 4 lines selected right now right?

Reply:
it is 3, because first is included and last excluded:
```python
In[1]:  l = list(range(1, 10))
In[2]:  i = 5
In[3]:  l[i-1:i+2]
Out[3]: [5, 6, 7]
```

Reply:
huh lol, I really need a coffee. Sorry for the noise.

Reply:
np :)